rest_tools/server/handler.py

```diff
@@ -14,24 +12,30 @@
 from collections import defaultdict
 from typing import Any, Dict, Union
 
-import rest_tools
 import tornado.escape
 import tornado.gen
 import tornado.httpclient
 import tornado.httputil
 import tornado.web
 from tornado.auth import OAuth2Mixin
 
+import rest_tools
+from .decorators import catch_error
+from .stats import RouteStats
 from .. import telemetry as wtt
 from ..utils.auth import Auth, OpenIDAuth
 from ..utils.json_util import json_decode
 from ..utils.pkce import PKCEMixin
-from .decorators import catch_error
-from .stats import RouteStats
 
 LOGGER = logging.getLogger('rest')
 
 
+def _log_auth_failed(e: Exception):
+    LOGGER.warning('failed auth')
```

Comment:
I think this should just be `info` level?

Reply:
sure thing